src/activities/mirrorgame/mirrorgame.js

```diff
@@ -0,0 +1,526 @@
+                   }}}}
+                else if(Light[k].uniquename === "lightpath"+i+j+row+column && Light[k].visible === true
+                        && row>i && column<j)
+                {
+                    for( a =0; a < items.rows;a++) {
+                        for( b = 0; b < items.columns; b++) {
+                           for( c = 0; c < count ; c++) {
+                        if(Light[c].uniquename === "lightpath"+a+b+i+j && i>a && j>b)
+                           {
+                            Light[c].visible = true;done=1
+                            currLight[i][j].push(Light[c])}
+                    }}}
+
+
+                }}if(done)break;}if(done)break;}
+                break
+          case 4:
+              while(currLight[i][j].length!==0)
+              {
+                  var x = currLight[i][j].pop()
+                  x.visible = false
+              }
+              for(row = 0; row < items.rows; row++){done = 0
+                  for(column = 0; column < items.columns; column++){
+                      for(k = 0; k < count; k++) {
+
+
+                 if(Light[k].uniquename === "lightpath"+row+column+i+j && Light[k].visible === true)
+                {
+                    for( a =0; a < items.rows;a++) {
+                        for( b = 0; b < items.columns; b++) {
+                           for( c = 0; c < count ; c++) {
+                        if(Light[c].uniquename === "lightpath"+a+b+i+j && row<i && column<j
+                                && a<i && b>j)
+                           {
+                            Light[c].visible = true;
+                            currLight[i][j].push(Light[c])
+                          }
+                        else if(Light[c].uniquename === "lightpath"+a+b+i+j && row>i && column<j
+                                && a<i && b<j)
+                           {
+                            Light[c].visible = true;
+                            currLight[i][j].push(Light[c])
+                        }
+                }}}
+                }
+                }if(done)break}if(done)break}
+                break
+    }
+    }
+
+}}
+}
+checkAnswer();
+}
+function checkAnswer()
+{
+    for( var a =0; a < items.rows;a++) {
+        for( var b = 0; b < items.columns; b++) {
+           for( var c = 0; c < items.rows ; c++) {
+               for(var d = 0;d < items.columns; d++){
+                   for(var e = 0;e < count;e++){
+                    for(var f = 0;f < num;f++){
+                        if(bulboff[f].uniquename === "bulboff"+a+b &&
+                                (Light[e].uniquename === "lightpath"+a+b+c+d||
+                                 Light[e].uniquename === "lightpath"+c+d+a+b)
+                               &&
+                                Light[e].visible === true)
+                        {
+                            bulboff[f].visible = false;
+                            bulbon[f].visible = true;
```

Comment:
this goes in pair with the comment: "you can probably have only one bulb array and lighten it when necessary (changing the image source)"